### DISCUSSION TITLE
Need guidance on storing all the indexes at custom location

### DISCUSSION THREAD

**Discussion by RajKKapadia** [alexklibisz/elastiknn#343] [Q&A] [answered]
Hello everyone,

I am running elasrticknn on my ubuntu server with following things

(1) creating a Dockerfile 
FROM docker.elastic.co/elasticsearch/elasticsearch:7.16.2
RUN elasticsearch-plugin install --batch https://github.com/alexklibisz/elastiknn/releases/download/7.16.2.0/elastiknn-7.16.2.0.zip

(2) running following command to build the image
docker build -t elastiknn-example .

(3) finally running following comand to start the docker
docker run -d -p 9200:9200 -p 9300:9300 -e "discovery.type=single-node" elastiknn-example

It works smooth until I restart the server.
When I restart the server all my indexes reset as well.

I hope I am clear on the issue, but feel free to ask anything.

Regards,
Raj Kapadia

**Comment by alexklibisz** [comment, marked as answer]
Hey Raj, I recommend reading these docs: https://www.elastic.co/guide/en/elasticsearch/reference/current/docker.html#docker-compose-file

You basically need to mount a volume at `/usr/share/elasticsearch/data` in order to persist data. The example docker-compose file does this in its `volumes` sections. I'm not sure what the corresponding docker command is. I would google something like "docker command to mount a volume"

**Comment by RajKKapadia** [comment]
Hello Alexklibisz,

I hope you are doing well.

I highly appreciate your help in this matter. I think I got the error, I
was using different location than

/usr/share/elasticsearch/data

I think using this will solve the problem, I am optimistic for this.

Again thank you very much for this prompt response.

Regards,
Raj Kapadia
